pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,261 @@
+	allCRDNameList, err := getAllAntreaCRDNames(k8sClient)
+	if err != nil {
+		return nil, err
+	}
+	// If the user-provided list of CRDs to upgrade is empty, upgrade all CRDs managed by Antrea.
+	if crdNameListToUpgrade.Len() == 0 {
+		crdNameListToUpgrade = allCRDNameList
+	} else {
+		// If the user-provided list of CRDs to upgrade is not empty, and it contains CRDs which are managed by Antrea,
+		// skip the CRDs which are not managed by Antrea.
+		for name := range crdNameListToUpgrade.Difference(allCRDNameList) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Skip CRD %v which is not created by Antrea.\n", name)
+		}
+		// Only upgrade the CRDs managed by Antrea.
+		crdNameListToUpgrade = crdNameListToUpgrade.Intersection(allCRDNameList)
+	}
+	return crdNameListToUpgrade, nil
+}
+
+// getCRDListToUpgrade is used to get the list of CRDs to upgrade.
+func getCRDListToUpgrade(cmd *cobra.Command, k8sClient client.Client, crdNameListToUpgrade sets.Set[string]) []*apiextv1.CustomResourceDefinition {
+	var crdsToUpgrade []*apiextv1.CustomResourceDefinition
+	for crdName := range crdNameListToUpgrade {
+		crd := &apiextv1.CustomResourceDefinition{}
+		if err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: crdName}, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to get CRD %q: %v.\n", crdName, err)
+			continue
+		}
+		// The CRD that has only one version does not need to be upgraded.
+		if len(crd.Spec.Versions) < 2 {
+			fmt.Fprintf(cmd.OutOrStdout(), "No need to upgrade CRD %q since it only has one version.\n", crdName)
+			continue
+		}
+		crdsToUpgrade = append(crdsToUpgrade, crd)
+	}
+	return crdsToUpgrade
+}
+
+// upgradeCRDObjects is used to upgrade the objects of a CRD.
+func upgradeCRDObjects(cmd *cobra.Command, k8sClient client.Client, crd *apiextv1.CustomResourceDefinition) error {
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   crd.Spec.Group,
+		Version: getCRDStorageVersion(crd),
+		Kind:    crd.Spec.Names.ListKind,
+	})
+	if err := k8sClient.List(context.TODO(), list); err != nil {
+		return err
+	}
+	if len(list.Items) == 0 {
+		return nil
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Upgrading %d CRD %q objects.\n", len(list.Items), crd.Name)
+	for _, obj := range list.Items {
+		copyObj := obj
+		err := retry.OnError(wait.Backoff{
+			Duration: time.Second,
+			Steps:    5,
+		}, func(err error) bool {
+			return validateErr(err) != nil
+		}, func() error {
+			return k8sClient.Update(context.TODO(), &copyObj)
+		})
+		if validateErr(err) != nil {
+			return err
+		}
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Successfully upgraded %d CRD %s objects.\n", len(list.Items), crd.Name)
+	return nil
+}
+
+// patchCRDStoredVersions is used to patch field status.storedVersion of a CRD.
+func patchCRDStoredVersions(k8sClient client.Client, crd *apiextv1.CustomResourceDefinition) error {
```

Comment:
I suppose user have to set one and only one version's `storage` as true when the CRD has multiple versions, so why do we need to patch the status.storedVersion of a CRD?

Reply:
We may a CRD which has more than one value in `status.storedVersion` in this case:
- create v1alpha1 of the CRD, and field of `storage` in v1alpha is set to `true`.
- create a CR of version v1alpha1, 
- add v1beta1 of the CRD, and field of `storage` in v1beta1 is set to `true` and `storage` in v1beta1 is set to `false`.
- create a CR of version v1beta1

We will get a CRD whose `status.storedVersion` has two values. 

User may have the objects of the CRD in multiple versions. After upgrading all objects, the `status.storedVersion` should only have the latest version value.

Reply:
Will status.storedVersion be managed and updated by K8s automatically? Have you tried to update all CRs to latest version and check after upgrade later?
cc @tnqn may have more ideas.